src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerChangeService.java

```diff
@@ -0,0 +1,291 @@
+        }
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                personName,
+                CompanyOfficerApi::getName,
+                Function.identity(),
+                ComparisonHelper::equals,
+                IndividualManagingOfficer::setPersonName
+        );
+
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                managingOfficerIndividualDto.getFormerNames(),
+                CompanyOfficerApi::getFormerNames,
+                Function.identity(),
+                ComparisonHelper::equalsFormerName,
+                IndividualManagingOfficer::setFormerNames
+        );
+
+        hasChange |= changeManager.compareAndBuildRightChange(
+                managingOfficerIndividualDto.getUsualResidentialAddress(),
+                ManagingOfficerDataApi::getResidentialAddress,
+                TypeConverter::addressDtoToAddress,
+                ComparisonHelper::equals,
+                IndividualManagingOfficer::setResidentialAddress
+        );
+
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                managingOfficerIndividualDto.getOccupation(),
+                CompanyOfficerApi::getOccupation,
+                IndividualManagingOfficer::setOccupation
+        );
+
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                managingOfficerIndividualDto.getRoleAndResponsibilities(),
+                CompanyOfficerApi::getOfficerRole,
+                Function.identity(),
+                ComparisonHelper::equals,
+                IndividualManagingOfficer::setRole
+        );
+
+        var nationalitySubmission = generateNationalityOtherField(managingOfficerIndividualDto.getNationality(),
+                managingOfficerIndividualDto.getSecondNationality());
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                nationalitySubmission,
+                CompanyOfficerApi::getNationality,
+                IndividualManagingOfficer::setNationalityOther
+        );
+
+        return hasChange ? managingOfficerChange : null;
+    }
+
+    //todo: add comment
+    private ManagingOfficerChange<CorporateManagingOfficer> covertManagingOfficerCorporateToChange(
+            ManagingOfficerCorporateDto managingOfficerCorporateDto) {
+        var managingOfficerChange = new CorporateManagingOfficerChange();
+
+        var officer = new CorporateManagingOfficer();
+
+        Pair<CompanyOfficerApi, ManagingOfficerDataApi> publicPrivateMoPair = publicPrivateMo.get(
+                managingOfficerCorporateDto.getChipsReference());
+
+        if (publicPrivateMoPair == null) {
+            ApiLogger.errorContext(SERVICE, NO_PAIR_FOUND, null);
+            return null;
+        }
+        ChangeManager<CorporateManagingOfficer, CompanyOfficerApi, ManagingOfficerDataApi> changeManager
+                = new ChangeManager<>(officer, publicPrivateMoPair);
+
+        if (publicPrivateMoPair.getRight() != null) {
```

Comment:
See comment above on raising an error instead of producing a change without an id.